src/scheduling/schedule_service.py

```diff
@@ -83,6 +83,15 @@ def create_job(self, script_name, parameter_values, incoming_schedule_config, us
         if not schedule_config.repeatable and date_utils.is_past(schedule_config.start_datetime):
             raise InvalidScheduleException('Start date should be in the future')
 
+        if schedule_config.endOption == 'on':
+            if schedule_config.start_datetime > schedule_config.endArg:
+                raise InvalidScheduleException('End date should be after start date')
+            if date_utils.is_past(schedule_config.endArg):
+                raise InvalidScheduleException('End date should be in the future')
```

Comment:
I think this check is redundant. We already check, that start time is in the future

Reply:
Good point, removed.